### PR TITLE
Added possibility to set any custom tracking parameters for 3rd party plugins

### PR DESCRIPTION
**Pull request by tsteur**
Eg needed for Custom Dimensions plugin  ( https://github.com/piwik/piwik/issues/9129 )

### CODE REVIEW

File: PiwikTracker.php [hunk 1]
```diff
@@ -304,6 +305,27 @@ public function clearCustomVariables()
         $this->eventCustomVar = array();
     }
 
+    /**
+     * Sets a custom tracking paramater. This is useful if you need to send any tracking parameters for a 3rd party
```
Comment (mattab) [inline PiwikTracker.php]: paramater -> parameter

File: PiwikTracker.php [hunk 2]
```diff
@@ -304,6 +305,27 @@ public function clearCustomVariables()
...
+     * plugin that is not shipped with Piwik itself. Please note that custom parameters are cleared after each
+     * tracking request.
+     *
+     * @param string $trackingApiParameter The name of the tracking API parameter, eg 'dimension1'
+     * @param string $value Custom variable value  The value that shall be sent for this tracking parameter.
```
Comment (mattab) [inline PiwikTracker.php]: Custom variable value -> Tracking Parameter value

File: PiwikTracker.php [hunk 3]
```diff
@@ -304,6 +305,27 @@ public function clearCustomVariables()
...
+     */
+    public function setCustomTrackingParameter($trackingApiParameter, $value)
+    {
+        $this->customParameters[$trackingApiParameter] = $value;
+    }
+
+    /**
+     * Clear / reset all previously set custom tracking parameters.
+     */
+    public function clearCustomTrackingParameter()
```
Comment (mattab) [inline PiwikTracker.php]: clearCustomTrackingParameter -> clearCustomTrackingParameters

**Comment by tsteur** [comment]
Applied changed and rebased.

**Comment by tsteur** [comment]
@mattab is it ok to merge this one?

**Comment by mattab** [comment]
Looks good to me

**Pull request merged**